src/scripts/shouldCallAPI.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+function shouldCallAPI(state, expiresInSeconds) {
```

Comment:
You're right that the implementation of `shouldCallAPI` is pretty custom, however I think this is particularly a powerful feature for this middleware to have.

I think we're better off not including this particular function with the middleware, but still leaving the _ability_ to have a `shouldCallAPI` function in your action:

```js
const mockAction = {
  type: 'MOCK_ACTION',
  shouldCallAPI: (state) => someCustomFunction(foo, bar, state.biz),
  callAPI: () => mockAPICall,
  payload
};
```

And we can abstract the `shouldCallAPI` reference in `callAPI`:

```js
if (!shouldCallAPI(getState())) {
  return Promise.reject(new Error('Not calling API: shouldCallAPI flag was set as false.'));
}
```

Passing the state through seems like something we should do, and anything else will be up to the user's implementation.